src/components/ContentHighlights/CurrentContentHighlightHeader.jsx

```diff
@@ -1,28 +1,76 @@
-import React from 'react';
+import React, { useState, useEffect, useContext } from 'react';
 import {
-  Button, ActionRow,
+  Button, ActionRow, Alert,
 } from '@edx/paragon';
-import { Add } from '@edx/paragon/icons';
-
+import { Add, Info } from '@edx/paragon/icons';
+import { EnterpriseAppContext } from '../EnterpriseApp/EnterpriseAppContextProvider';
 import { useContentHighlightsContext } from './data/hooks';
-import { BUTTON_TEXT, HEADER_TEXT } from './data/constants';
+import {
+  BUTTON_TEXT, HEADER_TEXT, MAX_HIGHLIGHT_SETS_PER_ENTERPRISE_CURATION, ALERT_TEXT,
+} from './data/constants';
 
 const CurrentContentHighlightHeader = () => {
+  const {
+    enterpriseCuration: {
+      enterpriseCuration: {
+        highlightSets,
+      },
+    },
+  } = useContext(EnterpriseAppContext);
   const { openStepperModal } = useContentHighlightsContext();
+  // Preliminiary logic for the header text given max sets reached
+  const [maxHighlightsReached, setMaxHighlightsReached] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const createNewHighlight = () => {
+    if (maxHighlightsReached) {
+      setShowAlert(true);
+    } else {
+      openStepperModal();
```

Comment:
[curious] should we get coverage on this line (e.g., mocking `openStepperModal` from `useContentHighlightsContext` and then `expect(openStepperModal).toHaveBeenCalledTimes(1))`?

Reply:
Plan to come back to this during the tabbed integration of highlight dashboard, will keep open as a reminder.